v3/src/components/graph/adornments/plotted-value/plotted-value.tsx

```diff
@@ -0,0 +1,162 @@
+import React, { useCallback, useEffect, useRef } from "react"
+import { autorun } from "mobx"
+import { select, Selection } from "d3"
+import { IPlottedValueModel } from "./plotted-value-model"
+import { INumericAxisModel } from "../../../axis/models/axis-model"
+import { useAxisLayoutContext } from "../../../axis/models/axis-layout-context"
+import { ScaleNumericBaseType } from "../../../axis/axis-types"
+import { useDataConfigurationContext } from "../../hooks/use-data-configuration-context"
+import { observer } from "mobx-react-lite"
+import { kPlottedValueType } from "./plotted-value-types"
+import { useGraphContentModelContext } from "../../hooks/use-graph-content-model-context"
+import { usePrevious } from "../../hooks/use-previous"
+
+import "./plotted-value.scss"
+
+interface IValueObject {
+  line?: Selection<SVGLineElement, unknown, null, undefined>
+  cover?: Selection<SVGLineElement, unknown, null, undefined>
+  valueLabel?: Selection<SVGTextElement, unknown, null, undefined>
+}
+
+interface IProps {
+  cellKey: Record<string, string>
+  containerId?: string
+  model: IPlottedValueModel
+  plotHeight: number
+  plotWidth: number
+  xAxis?: INumericAxisModel
+  yAxis?: INumericAxisModel
+}
+
+export const PlottedValue = observer(function PlottedValue (props: IProps) {
+  const {cellKey={}, containerId, model, plotHeight, plotWidth, xAxis, yAxis} = props
+  const graphModel = useGraphContentModelContext()
+  const layout = useAxisLayoutContext()
+  const dataConfig = useDataConfigurationContext()
+  const dataset = dataConfig?.dataset
+  const xScale = layout.getAxisScale("bottom") as ScaleNumericBaseType
+  const xAttrId = dataConfig?.attributeID("x")
+  const xAttrType = dataConfig?.attributeType("x")
+  const yScale = layout.getAxisScale("left") as ScaleNumericBaseType
+  const yAttrId = dataConfig?.attributeID("y")
+  const yAttrType = dataConfig?.attributeType("y")
+  const classFromKey = model.classNameFromKey(cellKey)
+  const valueRef = useRef<SVGGElement>(null)
+  const isVertical = useRef(!!(xAttrType && xAttrType === "numeric"))
+  const previousAttrTypes = usePrevious({ xAttrType, yAttrType})
+  // TODO: The offset value below shouldn't be needed once there's a method available for adding
+  // space to the top of the graph to accommodate the Plotted Value's UI elements.
+  const offsetTop = 20
+
+  const getCases = useCallback(() => {
+    const cases = xAttrId && isVertical
+      ? dataset?.getCasesForAttributes([xAttrId])
+      : yAttrId && !isVertical
+        ? dataset?.getCasesForAttributes([yAttrId])
+        : []
+    return cases
+  }, [dataset, isVertical, xAttrId, yAttrId])
+
+  const getCaseValues = useCallback(() => {
+    const cases = getCases()
+    const caseValues = cases && xAttrId
+      ? cases.map(c => dataset?.getNumeric(c.__id__, xAttrId)).filter(x => x) as number[]
+      : cases && yAttrId
+        ? cases.map(c => dataset?.getNumeric(c.__id__, yAttrId)).filter(x => x) as number[]
```

Comment:
Seems like this would strip out zeros. I think you want something like:
```suggestion
      ? cases.map(c => dataset?.getNumeric(c.__id__, xAttrId)).filter(x => isFinite(x)) as number[]
      : cases && yAttrId
        ? cases.map(c => dataset?.getNumeric(c.__id__, yAttrId)).filter(x => isFinite(x)) as number[]
```